### PR TITLE
Add Dutch translation (nl_NL)

### PR DESCRIPTION
**Pull request by hypothermic**
Added a translation for the Dutch language.

I've also enabled it in the module's `build.gradle`.

### CODE REVIEW

File: app/src/main/res/values-nl/strings.xml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="header_all_songs">Alle Nummers</string>
+    <string name="playlist">Afspeellijst</string>
+    <string name="header_create_playlist">Creëer afspeellijst</string>
+    <string name="playlist_auto">Slimme Afspeellijst</string>
+    <string name="header_add_queue_to_playlist">Voeg wachtwij toe aan afspeellijst</string>
+    <string name="playlist_auto_new">Nieuwe Slimme Afspeellijst</string>
+    <string name="playlist_auto_choose_by">Kies nummers met</string>
+    <string name="first_launch_title">Welkom bij Jockey</string>
+    <string name="action_read_privacy_policy">Open</string>
+    <string name="alert_privacy_policy_updated">Jockey\'s privacy policy is gewijzigd</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Does "Privacy policy" need to be translated?

File: app/src/main/res/values-nl/strings.xml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="action_refresh">Bibliotheek herladen</string>
+    <string name="action_sort">Sorteer...</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Can we replace the `...`'s with `&#8230;`? This will use the ellipses character instead, which looks slightly nicer and is consistent with the other translations. (Same thing applies in several other places throughout this file)

File: app/src/main/res/values-nl/strings.xml [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="action_sort_date_added">Sorteren op laatst toegevoegd</string>
+    <string name="action_sort_date_played">Sorteren op laatst afgespeeld</string>
+    <string name="action_disable_shuffle">Shuffle uitschakelen</string>
+    <string name="action_enable_shuffle">Shuffle inschakelen</string>
+    <string name="action_enable_repeat">Herhalen inschakelen</string>
+    <string name="action_disable_sleep_timer">Slaaptimer uitschakelen</string>
+    <string name="action_done">Klaar</string>
+    <string name="action_select">Selecteer</string>
+    <string name="action_clear_queue">Wachtrij wissen</string>
+    <string name="action_add_queue_to_playlist">Wachtrij toevoegen aan afspeellijst</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Should this have ellipses at the end?

File: app/src/main/res/values-nl/strings.xml [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="action_set_sleep_timer">Slaaptimer...</string>
+    <string name="action_export_playlist">Wachtrij opslaan als afspeellijst...</string>
+    <string name="widget_compact_name">Jockey Compact</string>
+    <string name="widget_wide_name">Jockey Wijd</string>
+    <string name="widget_square_name">Jockey Illustratie</string>
+    <string name="confirm_removed_rule">Regel verwijderd</string>
+    <string name="prompt_save_changes">Wijzigingen opslaan?</string>
+    <string name="prompt_discard_changes">Wijzigingen ongedaan maken?</string>
+    <string name="confirm_application_exit">Druk nog een keer om af te sluiten</string>
+    <string name="message_play_error_not_found">Jockey kon “%1$s” niet vinden</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: This probably needs to use the HTML entity code for the curly double-quotes:  `&#8220;%1$s&#8221;` Same thing in several strings after this one.

File: app/src/main/res/values-nl/strings.xml [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="add_shortcut">Herlaad de kleur van het snelkoppelings-icoon</string>
+    <string name="add_shortcut_already_matched">Het snelkoppelings-icoon is al ingesteld volgens je huidige thema</string>
+    <string name="add_shortcut_description">Wil je dat de kleur van Jockey\'s snelkoppling-icoon hetzelfde is als je huidige thema? Jockey zal herstarten en alle icoontjes zullen verwijderd worden.</string>
+    <string name="uses_aosp_header">Android Support Libraries</string>
+    <string name="uses_header">Jockey gebruikt de volgende open-source onderdelen:</string>
+    <string name="uses_aosp_detail">Support-v4 27.1.0, Appcompat-v7 27.1.0, RecyclerView 27.1.0, CardView 27.1.0, Design 27.1.0, Palette 27.1.0, Preference-v14 27.1.0, Room 1.1.0; gedistributeerd door het Android Open Source Project onder de Apache License, versie 2.0</string>
+    <string name="uses_exoplayer_detail">Versie 2.6.1; gedistributeerd door Google Inc. onder een Apache License, versie 2.0</string>
+    <string name="uses_libflac_detail">Version 1.3.1; distributed by the Xiph.Org Foundation under a BSD License</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Looks like "distributed by" didn't get translated here

File: app/src/main/res/values-nl/strings.xml [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="color_grey">Grijs</string>
+    <string name="color_red">Rood</string>
+    <string name="color_orange">Oranje</string>
+    <string name="color_yellow">Geel</string>
+    <string name="color_green">Groen</string>
+    <string name="color_cyan">Cyaan</string>
+    <string name="color_blue">Blauw</string>
+    <string name="color_purple">Paars</string>
+    <string name="color_black">Zwart</string>
+    <string name="color_teal">Teal</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Google Translate is saying that this should be translated as "Taling". Is that accurate?

File: app/src/main/res/values-nl/strings.xml [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="theme_dark">Donker</string>
+    <string name="theme_light">Licht</string>
+    <string name="theme_auto">Automatisch</string>
+    <string name="playlist_maximum_prefix">Sta niet meer toe dan</string>
+    <string name="playlist_maximum_suffix">nummers</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Looks like "Allow no more than 25 songs" gets literally translated as "Allow no more than 25 numbers"? Is that correct here?

File: app/src/main/res/values-nl/strings.xml [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="empty_directory">Deze map heeft geen mappen</string>
+    <string name="about_open_privacy_policy">Privacybeleid</string>
+    <string name="lastfm_header">Artiesteninformatie wordt aangeboden door</string>
+    <string name="button_play_description">Afspeelknop</string>
+    <string name="button_skip_description">Volgende nummer knop</string>
+    <string name="button_previous_description">Vorige nummer knop</string>
+    <string name="description_album_art">Album Art</string>
+    <string name="player_notification_channel_name">Muziek Afspelen</string>
+    <string name="hint_playlist_name">Afspeellijst naam</string>
+    <string name="time_in_min_format">%1$d min</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: "Min" here is short for "Minutes," which translates to "Notulen" according to Google Translate. Do you know what a similar abbreviation would be for this?

File: app/src/main/res/values-nl/strings.xml [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="uses_timber_detail">Versie 1.1; gedistributeerd door Triggertrap Ltd (Neil Davies) onder een MIT License</string>
+    <string name="about_description">Jockey is een muziekspeler voor Android 4.1 en hoger, gefocused op simpliciteit, snelheid en design gedistributeerd onder een Apache 2.0 licentie.</string>
+    <string name="empty_included_dirs_detail">Jockey kan beperkt worden om in alleen specifieke mappen te doorzoeken, druk op de “+” knop om een map in te sluiten.</string>
+    <string name="empty_excluded_dirs_detail">Jockey kan beperkt worden om sommige mappen over te slaan, druk op de “+” knop om een map uit te sluiten.</string>
+    <string name="url_privacy_policy">https://www.marverenic.com/Jockey/privacy-policy.html</string>
+    <string name="first_launch_detail">"<![CDATA[Door deze applicatie te gebruiken ga je akkoord met Jockey\'s         <a href="https://www.marverenic.com/Jockey/privacy-policy.html">privacybeleid</a>.]]>     "</string>
+    <string name="pref_resume_on_connect">Speel af als een hoofdtelefoon wordt verbonden</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: This header is actually getting truncated on my Pixel XL. Is there a way to rephrase this to be more concise? It doesn't have to translate back to "Play when plugging in headphones", just as long as it's able to describe the concept of "Music will be played when you plug in any pair of headphones".

File: app/src/main/res/values-nl/strings.xml [hunk 10]
```diff
@@ -0,0 +1,259 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="header_playlists">Afspeellijsten</string>
+    <string name="header_songs">Nummers</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Is this the right translation? Google Translate is saying this means "Numbers"

File: app/src/main/res/values-nl/strings.xml (hunk 9 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: This is the shortest grammatically correct sentence I can think of.

I'll change it in the next commit to "Afspelen als headset verbonden", which is not grammatically correct, but the concept of "Music will be played when you plug in any pair of headphones" will still be clear.

File: app/src/main/res/values-nl/strings.xml (hunk 10 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Well, the literal translation of "Nummers" is indeed "Numbers", but we also use it for "Songs".

"Songs" gets translated to either "Nummers" or "Liedjes"

File: app/src/main/res/values-nl/strings.xml (hunk 8 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Again, Google Translate is wrong here...

"Notulen" means "notes" in English (as in, taking notes in class)

File: app/src/main/res/values-nl/strings.xml (hunk 7 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: The correct translation of "Allow no more than 25 songs" is "Sta niet meer toe dan 25 nummers".

I do not see any problem here?

File: app/src/main/res/values-nl/strings.xml (hunk 6 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: "Teal" is not used in the Dutch language, so it is spoken as the English word.

Google Translate is wrong here, a "taling" is a bird/duck and not a color.

File: app/src/main/res/values-nl/strings.xml (hunk 1 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: We use both "Privacybeleid" and "Privacy policy" (the latter being taken from English, obviously), but to be consistent with the rest I'll translate this one as well.

File: app/src/main/res/values-nl/strings.xml (hunk 2 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Fixed.

File: app/src/main/res/values-nl/strings.xml (hunk 3 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Fixed.

File: app/src/main/res/values-nl/strings.xml (hunk 4 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Fixed.

File: app/src/main/res/values-nl/strings.xml (hunk 5 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Fixed.

File: app/src/main/res/values-nl/strings.xml [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="uses_glide_detail">Versie 3.7.0; gedistributeerd door Google, Inc. onder een BSD licensie</string>
+    <string name="uses_vert_slider_detail">Versie 0.5.2; gedistributeerd door Haruki Hasegawa onder een Apache License, versie 2.0</string>
+    <string name="uses_fastscroll_detail">Versie 1.0.16; gedistributeerd door Tim Malseed onder een Apache License, versie 2.0</string>
+    <string name="uses_seek_arc_detail">Versie 1.1; gedistributeerd door Triggertrap Ltd (Neil Davies) onder een MIT License</string>
+    <string name="uses_timber_detail">Versie 1.1; gedistributeerd door Triggertrap Ltd (Neil Davies) onder een MIT License</string>
+    <string name="about_description">Jockey is een muziekspeler voor Android 4.1 en hoger, gefocused op simpliciteit, snelheid en design gedistributeerd onder een Apache 2.0 licentie.</string>
+    <string name="empty_included_dirs_detail">Jockey kan beperkt worden om in alleen specifieke mappen te doorzoeken, druk op de “+” knop om een map in te sluiten.</string>
+    <string name="empty_excluded_dirs_detail">Jockey kan beperkt worden om sommige mappen over te slaan, druk op de “+” knop om een map uit te sluiten.</string>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Looks like the quotes here still need to get converted to the HTML codes.

File: app/src/main/res/values-nl/strings.xml [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+    <string name="pref_sls_description_enabled_but_uninstalled">"<![CDATA[        Om verder te gaan met scrobbelen naar Last.fm, Simple Last.fm Scrobbler moet geinstalleerd blijven. Je         afspeelhistorie zal dan niet worden gedeeld. Als je niet langer wil scrobbelen kan je dit uitzetten.         <br/><br/>         Jockey kan je afspeelhistorie online plaatsen naar services zoals Last.fm door te integreren met         een onafhankelijke applicatie. Download Simple Last.fm Scrobbler van de         Google Play Store.         <br/><br/>         Je kan meer informatie vinden over Simple Last.fm Scrobbler vinden        <a href="https://github.com/tgwizard/sls/">op zijn website</a>.]]>     "</string>
+    <string name="equalizer_conflict_note">Er is een andere equalizer app geinstalleerd op dit apparaat. Dit kan problemen veroorzaken met Jockey\'s equalizer. Schakel Jockey\'s equalizer uit om de systeem-wijde equalizer te gebruiken.</string>
+</resources>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: I just remembered there's one more thing this file will need, and it's really subtle (and quite frankly, I have no idea how to even begin researching this aside from talking to someone who knows the language).

In English, when sorting music, books, and other works based on their titles, we ignore the articles "The", "A", and "An" when they appear at the start of the title. So instead of placing "The Bridge Between" in the same place as songs with titles that begin with 'T', we place them in the same place as songs with titles that start with the letter 'B'. Does Dutch have a similar convention? If it does, can you add this array to the end of this file?

```xml
    <!-- Prefixes that are ignored when sorting titles -->
    <string-array name="ignored_title_prefixes" tools:ignore="InconsistentArrays">
        <item>The</item>
        <item>A</item>
        <item>An</item>
    </string-array>
```

If there is no such convention, you can just define an empty `<string-array ...>` with no `<item>`'s which will disable this sorting behavior (Currently, it will fall back and ignore English articles if they appear at the beginning of a title, which might not be the right thing to do).

File: app/src/main/res/values-nl/strings.xml (hunk 11 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: Ah thanks, I use the Translations Editor in Android Studio and it shows `"+"` instead of `&#8220;+&#8221;` so I was unaware of this. (same thing with `"%1$s"`)

File: app/src/main/res/values-nl/strings.xml (hunk 12 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: There is no national standard for music, however from my experience with keeping a vinyl/CD collection I can say that most people ignore the prefixes.

I've added the piece of code to the end of the file and committed it, but Android Studio displays an error: "Namespace 'tools' is not bound". Not sure if that's just my configuration or if it's incorrect.

---

Also, when sorting music by the *artist*, we ignore the following prefixes (capitalisation ignored):
```xml
<string-array name="ignored_title_prefixes" tools:ignore="InconsistentArrays">
        <item>Van</item>
        <item>De</item>
        <item>In</item>
        <item>Ten</item>
        <item>Van den</item>
        <item>Van der</item>
</string-array>
```
and surnames starting with `IJ` are sorted under the `Y`

---

References: [1](https://sec-florijn.learningmatters.nl/informatiemanagement/p1/alfabetisch-rangschikken), [2](https://nl.wikipedia.org/wiki/Alfabetische_volgorde)

**Comment by hypothermic** [comment]
This is my first time using GitHub's "change request" thing so I'm not sure if you're getting notifications or not. Leaving this comment here to make sure you got the notification.

File: app/src/main/res/values-nl/strings.xml [hunk 13]
```diff
@@ -0,0 +1,272 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: To fix the compiler error about `tools`, you need to tweak one thing here (the translations editor is most likely completely unaware of this).

```suggestion
<resources xmlns:tools="http://schemas.android.com/tools">
```

File: app/src/main/res/values-nl/strings.xml [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+    <!-- Prefixes that are ignored when sorting titles -->
+    <string-array name="ignored_title_prefixes" tools:ignore="InconsistentArrays">
+        <item>De</item>
+        <item>Het</item>
+        <item>Een</item>
+        <item>Ten</item>
+        <item>d'</item>
+        <item>'t</item>
+        <item>'s</item>
+        <item>d' n</item>
```
Comment (marverenic) [inline app/src/main/res/values-nl/strings.xml]: Quick question about the last four strings here. How do they appear in titles? Are they glued to other words? Or do they appear with a space afterwards?

The current logic here is to filter these out as entire words. So "The Bridge Between" gets grouped with B's, but "Theater Between" will get grouped with the T's. There's an extremely high chance that this approach breaks down in other languages and that the sorting logic needs to get updated, but that's not something we need to fix right this second in this PR.

File: app/src/main/res/values-nl/strings.xml (hunk 12 above)
Comment (marverenic) [reply app/src/main/res/values-nl/strings.xml]: Oh. That's awesome to know. There isn't a way to designate that logic right now (since English doesn't have similar rules), but that's definitely something that can be built in with this knowledge.

That said, the surname rule will be virtually impossible to code in since there isn't a good way to figure out whether a word is a surname or not. Let me know if there's a heuristic that might work as an approximation for that rule, or if we should just ignore it for now.

(Also, GitHub is really weird about replying in threads during a review, so this comment will appear as part of the review, but to reply to it you have to scroll back up to the original thread. No idea why they did it that way)

File: app/src/main/res/values-nl/strings.xml (hunk 14 above)
Comment (hypothermic) [reply app/src/main/res/values-nl/strings.xml]: They all appear with a space afterwards except for `d'`.

You could say they're abbrevations, for example:
`Het Zwarte Paard` --> `'t Zwarte Paard`

and since `Het` is a prefix, we should ignore `'t` too.

Also, the circleci build has failed due to non-escpaed `'`'s, I'll repair it in the next commit.

**Comment by hypothermic** [comment]
> Let me know if there's a heuristic that might work as an approximation for that rule, or if we should just ignore it for now

It's not important, might be good for a first issue.

As you said, figuring out what's a surname or not is a difficult task, especially if the artist name has nothing to do with the person (ex. "AC/DC" or "The Eagles")

**Comment by hypothermic** [comment]
I have too much free time on my hands currently so I'd be happy to translate the Play listing or any other apps that you wish.
